test/listener_test.coffee

```diff
@@ -17,6 +17,8 @@ describe 'Listener', ->
     @robot =
       # Why is this part of the Robot object??
       Response: Response
+      runPrelistenHooks: (listener, response) ->
+        listener.callback(response)
```

Comment:
We're stubbing out response, so I have worsened the problem for now. If we like the direction I would try to refactor the hook handling out of the listener and let the robot handle it all, and this could go away.